### PR TITLE
Vmo 5442 flow form fields

### PR DESCRIPTION
**Pull request by safydy-r**
Voto5 PR: https://bitbucket.org/voto/voto5/pull-requests/2307/vmo-5442-flow-form-fields

Merge plan:
- merge voto5's vmo-4597 before merging this to `master`

TODO:
- fix eslint related to changed lines

### CODE REVIEW

File: src/components/interaction-designer/flow-editors/FlowEditor.vue [hunk 1]
```diff
@@ -1,48 +1,55 @@
 <template>
   <div class="flow-editor">
-    <h3 class="no-room-above">
+    <h3 class="no-room-above" v-if="flowHeader">
       {{ flowHeader | trans }}
     </h3>
 
-    <fieldset :disabled="!isEditable && sidebar">
+    <fieldset :disabled="!isEditable && isOnSmallContainer">
```
Comment (safydy-r) [inline src/components/interaction-designer/flow-editors/FlowEditor.vue:7]: renamed `sidebar` into `isOnSmallContainer`

File: src/views/NewFlow.vue [hunk 2]
```diff
@@ -57,22 +60,42 @@ class NewFlow extends Vue {
...
     await this.flow_addBlankFlow()
   }
 
-  async created(): Promise<void> {
+  async beforeCreate(): Promise<void> {
     const {$store} = this
 
-    forEach(store.modules, (v, k) =>
-      !$store.hasModule(k) && $store.registerModule(k, v))
+    forEach(store.modules, (v, k) => !$store.hasModule(k) && $store.registerModule(k, v))
```
Comment (safydy-r) [inline src/views/NewFlow.vue:84]: had to extract module registration in `beforeCreate` to fix refresh on `flows/new` route

Comment (safydy-r) [reply src/views/NewFlow.vue:84]: otherwise we had something like here https://github.com/vuejs/vuex/issues/1622#issuecomment-536960343

File: src/components/interaction-designer/flow-editors/FlowEditor.vue [hunk 3]
```diff
@@ -76,7 +79,26 @@ const validationVuexNamespace = namespace('validation')
 class FlowEditor extends mixins(Lang) {
   @Prop() readonly flow!: IFlow
   @Prop({default: 'flow-builder.edit-flow'}) readonly flowHeader!: string
-  @Prop({default: true}) readonly sidebar!: boolean
+  @Prop({default: true}) readonly isOnSmallContainer!: boolean
```
Comment (bulatgab) [inline src/components/interaction-designer/flow-editors/FlowEditor.vue:88]: I like how you renamed it

File: src/components/interaction-designer/flow-editors/FlowEditor.vue [hunk 4]
```diff
@@ -75,8 +84,28 @@ const validationVuexNamespace = namespace('validation')
...
+    } if (!this.hasDefaultInteractionTimeout) {
+      return 'col-xl-6 col-lg-6 col-md-12 col-sm-12'
+    } else {
+      return 'col-12'
+    }
+  }
+
+  get otherRowsClass(): string {
+    return this.isOnSmallContainer ? 'col-12' : 'col-xl-6 col-lg-6 col-md-12 col-sm-12'
+  }
```
Comment (bulatgab) [inline src/components/interaction-designer/flow-editors/FlowEditor.vue:108]: I spent quite some time trying to solve this whole styling thing. I'm glad you succeeded 👏

File: builder.config.json [hunk 5]
```diff
@@ -346,7 +351,13 @@
...
       "USSD",
       "IVR"
-    ]
+    ],
+    "appWideInteractionTimeout": null,
+    "title": {
+      "createFlow": "flow-builder.create-flow",
+      "editFlow": "flow-builder.edit-flow",
+      "importFlow": "flow-builder.import-flow"
+    }
```
Comment (bulatgab) [inline builder.config.json:360]: I'm glad you undid my `showValidationErrorsOnlyAfterAttemptToPublishTree` config.
It was ugly. And it added unnecessary complexity.
Now both the community version and the embedded app will have the same behaviour (the errors will be shown only after the user tries to publish). And that's fine (I was trying to make it configurable, but no one said that it should be in the first place). Thanks!

Comment (bulatgab) [reply builder.config.json:360]: I also realize now that the AC was talking about validation errors on the `NewFlow` window and the flow creation action as a trigger for them. (And _not_ about the flow/block validation errors in the interaction designer and the publishing action as a trigger).

File: src/views/InteractionDesigner.vue [hunk 6]
```diff
@@ -144,11 +144,13 @@ export default {
     },
   },
 
-  created() {
+  async beforeCreate() {
     const {$store} = this
 
     forEach(store.modules, (v, k) => !$store.hasModule(k) && $store.registerModule(k, v))
+  },
```
Comment (bulatgab) [inline src/views/InteractionDesigner.vue:151]: out of curiosity, why was this part moved to `beforeCreate()`?

**Review by bulatgab — APPROVED** [review]

**Review by object-required — APPROVED** [review]

File: src/components/interaction-designer/flow-editors/FlowEditor.vue (hunk 4 above)
Comment (safydy-r) [reply src/components/interaction-designer/flow-editors/FlowEditor.vue:108]: yeah, sometimes it's quite difficult to handle the grid for different container size. I always use bootstrap grid to deal with such AC. https://getbootstrap.com/docs/4.0/layout/grid/#grid-options

File: src/components/interaction-designer/flow-editors/FlowEditor.vue (hunk 3 above)
Comment (safydy-r) [reply src/components/interaction-designer/flow-editors/FlowEditor.vue:88]: Thanks ;)

File: builder.config.json (hunk 5 above)
Comment (safydy-r) [reply builder.config.json:360]: No worries, sorry if there were some missed explanation from the ticket. Happy we made this together ;)
